File: Medium/1535. Find the Winner of an Array Game/1535. Find the Winner of an Array Game.py
from typing import List


class Solution:
    def getWinner(self, arr: List[int], k: int) -> int:
        current = arr[0]
        win_count = 0

        for i in range(1, len(arr)):
            if arr[i] > current:
                current = arr[i]
                win_count = 1
            else:
                win_count += 1

            if win_count == k:
                return current

        if k > len(arr):
            return max(arr)

        return current

File: Medium/1535. Find the Winner of an Array Game/1535. Find the Winner of an Array Game.rs
impl Solution {
    pub fn get_winner(arr: Vec<i32>, k: i32) -> i32 {
        let mut current = arr[0];
        let mut win_count = 0;

        for i in 1..arr.len() {
            if arr[i] > current {
                current = arr[i];
                win_count = 1;
            } else {
                win_count += 1;
            }

            if win_count == k {
                return current;
            }
        }

        if k > arr.len() as i32 {
            return *arr.iter().max().unwrap();
        }

        current
    }
}
